g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/dao/impl/LocalSourceDaoImpl.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.l10n.source.dao.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.vmware.l10n.record.dao.SqlLiteDao;
+import com.vmware.l10n.source.dao.SourceDao;
+import com.vmware.l10n.utils.SourceUtils;
+import com.vmware.vip.common.constants.ConstantsChar;
+import com.vmware.vip.common.constants.ConstantsFile;
+import com.vmware.vip.common.constants.ConstantsKeys;
+import com.vmware.vip.common.exceptions.VIPResourceOperationException;
+import com.vmware.vip.common.i18n.dto.SingleComponentDTO;
+import com.vmware.vip.common.i18n.resourcefile.LocalJSONReader;
+import com.vmware.vip.common.i18n.resourcefile.ResourceFilePathGetter;
+import com.vmware.vip.common.l10n.source.dto.ComponentMessagesDTO;
+import com.vmware.vip.common.utils.SortJSONUtils;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+@Repository
+@Profile(value="bundle")
+public class LocalSourceDaoImpl implements SourceDao {
+    private static Logger LOGGER = LoggerFactory.getLogger(LocalSourceDaoImpl.class);
+
+    @Autowired
+    private SqlLiteDao sqlLite;
+
+    /**
+     * the path of local resource file,can be configed in spring config file
+     **/
+    @Value("${source.bundle.file.basepath}")
+    private String basepath;
+
+	@Override
+	public String getFromBundle(SingleComponentDTO singleComponentDTO) {
+		String result = "";
+		String component = singleComponentDTO.getComponent();
+		if (StringUtils.isEmpty(component)) {
+			component = ConstantsFile.DEFAULT_COMPONENT;
+			singleComponentDTO.setComponent(component);
+		}
+		String filepath = ConstantsFile.L10N_BUNDLES_PATH
+				+ ResourceFilePathGetter
+						.getProductVersionConcatName(singleComponentDTO)
+				+ ConstantsChar.BACKSLASH
+				+ component
+				+ ConstantsChar.BACKSLASH
+				+ ResourceFilePathGetter
+						.getLocalizedJSONFileName(ConstantsKeys.LATEST);
+		LOGGER.info("Read content from file: {}{}",  basepath, filepath);
```

Comment:
all info log in dao layer should be changed to debug log, to avoid frequently log update and large log file;
all error log should be thrown to upper layer to handle.
please have a global check.

Reply:
Same as above. This is out of scope of this PR. This PR only focus on DAO layer.

Reply:
This file is existing before this PR. It's renamed from SourceDaoImpl.java

Reply:
changed to debug level.